### PR TITLE
fixes #1072 - sort by name

### LINKED ISSUE #1072: Link for alphabetic display of wiki index
Hi, this is a [first-timers-only issue](https://github.com/publiclab/plots2/projects/2). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

## The problem

As of #1049, we now have a great alphabetic display of wiki pages, at:

https://publiclab.org/wiki/?order=alphabetic

Which will be live shortly. But theres no link to get there from https://publiclab.org/wiki/

There are links for sorting by popularity and others:

screen shot 2016-12-06 at 5 25 19 pm

## Solution

We need to modify this template file:

https://github.com/publiclab/plots2/blob/master/app/views/wiki/index.html.erb#L18-L21

In order to show the "alphabetic" option, similarly to how the others work, we need a conditional like:

```ruby
  <% elsif params[:order] == "alphabetic" %>

    <h2><%= t('wiki.index.popular_wiki_pages') %> <small><%= t('wiki.index.collaborative_documentation') %></small></h2>
    <p>&nbsp;<a style="margin-bottom:4px;" href="/wiki/new" class="btn btn-default"><i class="fa fa-plus-circle"></i> <%= t('wiki.index.create_wiki_page') %></a> <span style="margin-top:4px;" class="pull-right"><a href="/wiki/"><%= t('wiki.index.by_recency') %></a> | <span style="color:#aaa"><%= t('wiki.index.by_popularity') %></span> | <a href="/wiki/liked"><%= t('wiki.index.by_number_of_likes') %></a></span></p>
```

But note how hard to read this all is. Perhaps we should:

1. break these up into multiple lines
2. use a different structure of independent `if/else` statements so that each link need only appear once

For example, each one could be like:

```ruby
<% if params[:order] == 'alphabetic' %>
  alphabetic
<% else %>
  <a href="/wiki/?order=alphabetic">alphabetic</a>
<% end %>
```

Try doing this for each of these four: recency, popularity, # of likes, alphabetic.

And please ask if you need help!

## Steps to Fix

- [x] claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] set up a repository locally following the README instructions, and make sure that all tests pass
- [ ] try to fix the issue following the steps above, but even before you're done, you can:
- [ ] commit your changes and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] alert someone via the developers list (see below) to have your pull request merged. This may take a few extra steps depending on when you do it, but we'll help you out!

Please email the developers list (see https://publiclab.org/developers) if you have questions, and take a look at [our first-timers landing page for more information](https://github.com/publiclab/plots2/projects/2)!

### PR DESCRIPTION
**Pull request by Mcosta02**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue
* [x] if possible, multiple commits squashed if they're smaller changes
* [x] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

fixes #1072

### CODE REVIEW

File: app/views/wiki/index.html.erb [hunk 1]
```diff
@@ -10,27 +10,50 @@
...
+      <% if params[:action] == "index" && params[:order] !="alphabetic" %>
+        <span style="color:#aaa"><%= t('wiki.index.by_recency') %></span> |
+      <%else%>
+        <a href="/wiki/"><%= t('wiki.index.by_recency') %></a> |
+      <% end %>
 
-  <% else %>
+      <% if params[:order] == "alphabetic" %>
+        <span style="color:#aaa"><%= t('wiki.index.by_alphabetic') %></span> |
+      <%else%>
```
Comment (jywarren) [inline app/views/wiki/index.html.erb]: This looks good, but if you could please observe standard spacing:

`<% else %>`

that really helps us keep our codebase tidy and readable. Thank you!

File: app/views/wiki/index.html.erb [hunk 2]
```diff
@@ -10,27 +10,50 @@
 
     <h3 style="margin-top:0;"><%= raw t('wiki.index.wiki_pages_by', :url1 => "/people/"+@user.name, :user => @user.name, :note_length => @user.notes.length, :time => time_ago_in_words(@user.notes.first.created_at)) %></h3>
 
-  <% elsif params[:action] == "liked" %>
+  <% end %>
 
-    <h2><%= t('wiki.index.highly_liked_wiki_pages') %> <small><%= t('wiki.index.collaborative_documentation') %></small></h2>
-    <p>&nbsp;<a style="margin-bottom:4px;" href="/wiki/new" class="btn btn-default"><i class="fa fa-plus-circle"></i> <%= t('wiki.index.create_wiki_page') %></a> <span style="margin-top:4px;" class="pull-right"><a href="/wiki/"><%= t('wiki.index.by_recency') %></a> | <a href="/wiki/popular"><%= t('wiki.index.by_popularity') %></a> | <span style="color:#aaa"><%= t('wiki.index.by_number_of_likes') %></span></span></p>
+  <h2 style="font-family:Junction Light;"><%= t('wiki.index.public_lab_wiki') %> <small><%= t('wiki.index.collaborative_documentation') %></small></h2>
```
Comment (jywarren) [inline app/views/wiki/index.html.erb]: Could this also be broken up over multiple lines and indented tidily? Thank you!

**Comment by jywarren** [comment]
I made a couple requests -- but could you also upload a screenshot of what you see on the page, rendered, so we can be sure? Thank you!!

**Comment by Mcosta02** [comment]
1072_sort_by recency
1072_sort_by_name
1072_sort_by_popularity
1072_sort_by_like

**Comment by Mcosta02** [comment]
Yes, no problems. Follow in attachment. Let me know any problem.
Thank you.

**Comment by jywarren** [comment]
Hi -- this looks perfect; merging now. If you're looking for a new challenge, please take a look at our `help-wanted` list: https://github.com/publiclab/plots2/labels/help-wanted

Thanks!

**Pull request merged**